### PR TITLE
feat(motions): add bw

### PR DESCRIPTION
**Pull request by yuu-ux**
# overview
Added key sequence support for dw.

# Potential Issues
In nvim, `dw` on `hello\nworld` deletes only `hello` (it does not delete the newline). In this PR, however, the current implementation deletes `hello\n`.

I originally tried to match nvim by adjusting the selection with conditional logic: if the selected text contained a newline, move the selection end back to just before the newline; if it contained tabs/spaces, move it to the end of the whitespace run. With that approach, I hit a bug where deletion did not run at all when the selected range contained a newline, and I wasn’t able to resolve it in a reasonable time. So I’m sending this PR with the simpler behavior for now.

If strict nvim compatibility is required, I can follow up and fix the newline behavior.
Since I'm weak in English, I'm using a translation app. Please let me know if there are any unclear English sentences.

### CODE REVIEW

File: src/glide/browser/base/content/motions.mts [hunk 1]
```diff
@@ -180,11 +180,28 @@ export function select_motion(
...
+        editor.selectionController.characterMove(false, false);
+        editor.selectionController.characterMove(true, true);
+      }
+
+      const starting_cls = text_obj.cls(current_char(editor));
+      if (is_eof(editor)) {
+        break;
+      }
+      forward_word(editor, false, "visual");
+      if (selection_has_cls_white_space(editor) || text_obj.cls(current_char(editor)) !== starting_cls) {
```
Comment (RobertCraigie) [inline src/glide/browser/base/content/motions.mts]: for what its worth, I'm surprised the `text_obj.cls(current_char(editor)) !== starting_cls` check didn't fix the `hello\nworld` case you mention in the description?

Comment (RobertCraigie) [reply src/glide/browser/base/content/motions.mts]: for the EOL case in particular, did you try the `is_eol` helper? https://github.com/glide-browser/glide/blob/c09d6747536e7001cb1b610e4ad30702b8da37a0/src/glide/browser/base/content/motions.mts#L603

File: src/glide/browser/base/content/test/motions/browser_words.ts [hunk 2]
```diff
@@ -100,6 +100,46 @@ add_task(async function test_normal_diw() {
...
+    await set_selection(0);
+    await test_edit("dw", "world", 0, "w");
+
+    await set_text("hello?.world", "dw on punctuation run deletes punctuation up to next word");
+    await set_selection(5);
+    await test_edit("dw", "helloworld", 5, "w");
+
+    await set_text("h h h", "dw deletes a single-letter word + following space");
+    await set_selection(2);
+    await test_edit("dw", "h h", 2, "h");
```
Comment (RobertCraigie) [inline src/glide/browser/base/content/test/motions/browser_words.ts]: nit: you can also set the expected char to make the test easier to read, and I also like to avoid duplicate chars for the same reason
```suggestion
    await set_selection(5, "?");
    await test_edit("dw", "helloworld", 5, "w");

    await set_text("h j k", "dw deletes a single-letter word + following space");
    await set_selection(2, "j");
    await test_edit("dw", "h k", 2, "k");
```

File: src/glide/browser/base/content/motions.mts (hunk 1 above)
Comment (yuu-ux) [reply src/glide/browser/base/content/motions.mts]: Thank you for your comment. Sorry for the delay, but I have also been able to handle cases including line breaks, so could you please check it?
https://github.com/glide-browser/glide/pull/126/commits/295d46fd1c8ae68202461ab17bb13a5828150b7a
